# Predicting Diabetes

In [20]:
from pathlib import Path
import pandas as pd
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')

In [21]:
data = Path('../Resources/diabetes.csv')
df = pd.read_csv(data)
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

## Separate the Features (X) from the Target (y)

## Split our data into training and testing

In [22]:
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, y, random_state=1, stratify=y)
X_train.shape

(576, 8)

## Create a Logistic Regression Model

In [26]:
from sklearn.linear_model import LogisticRegression
db_classifier = LogisticRegression(solver='lbfgs', max_iter=200, random_state=5)
db_classifier

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=200,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=5, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

## Fit (train) or model using the training data

In [27]:
db_classifier.fit(X_train, Y_train)

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=200,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=5, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

## Score the model using the test data

In [28]:
print("Training Data Score is: " + f'{db_classifier.score(X_train, Y_train)}')
print("Testing Data Score is: " + f'{db_classifier.score(X_test, Y_test)}')

Training Data Score is: 0.7829861111111112
Testing Data Score is: 0.7760416666666666


## Make predictions

In [29]:
Result = pd.DataFrame({'Prediction': db_classifier.predict(X_test), 'Actual': Y_test})
Result[Result['Actual']!=Result['Prediction']].count()

Prediction    43
Actual        43
dtype: int64

In [30]:
Result.count()

Prediction    192
Actual        192
dtype: int64

43 cases out of 192 samples didnt have the actual and predition matching. indicating 74% model accuracy.

In [31]:
from sklearn.metrics import confusion_matrix
c_matrix=pd.DataFrame(confusion_matrix(Y_test, Result['Prediction']), columns=['Predicted_No', 'Predicted_Yes'])
c_matrix.index=['Actual_No', 'Actual_Yes']
c_matrix

Predicted_No  Predicted_Yes
Actual_No            113             12
Actual_Yes            31             36

In [32]:
from sklearn.metrics import classification_report
target_names = ["Daibetic", "non_Daibetic"]
print(classification_report(Y_test, Result['Prediction'], target_names=target_names))

              precision    recall  f1-score   support

    Daibetic       0.78      0.90      0.84       125
non_Daibetic       0.75      0.54      0.63        67

    accuracy                           0.78       192
   macro avg       0.77      0.72      0.73       192
weighted avg       0.77      0.78      0.77       192

